src/Symfony/Component/AssetMapper/MappedAsset.php

```diff
@@ -0,0 +1,178 @@
+        }
+
+        return $this->content;
+    }
+
+    public function getDigest(): string
+    {
+        if (!isset($this->digest)) {
+            throw new \LogicException('Cannot get digest: it has not been set on the asset.');
+        }
+
+        return $this->digest;
+    }
+
+    public function isPredigested(): bool
+    {
+        return $this->isPredigested;
+    }
+
+    public function getMimeType(): string
+    {
+        if (!isset($this->mimeType)) {
+            throw new \LogicException('Cannot get mime type: it has not been set on the asset.');
+        }
+
+        return $this->mimeType;
+    }
+
+    /**
+     * @return AssetDependency[]
+     */
+    public function getDependencies(): array
+    {
+        return $this->dependencies;
+    }
+
+    public function setPublicPath(string $publicPath): void
+    {
+        if (isset($this->publicPath)) {
+            throw new \LogicException('Cannot set public path: it was already set on the asset.');
+        }
+
+        $this->publicPath = $publicPath;
+    }
+
+    public function setSourcePath(string $sourcePath): void
+    {
+        if (isset($this->sourcePath)) {
+            throw new \LogicException('Cannot set source path: it was already set on the asset.');
+        }
+
+        $this->sourcePath = $sourcePath;
+    }
+
+    public function setDigest(string $digest, bool $isPredigested): void
+    {
+        if (isset($this->digest)) {
+            throw new \LogicException('Cannot set digest: it was already set on the asset.');
+        }
+
+        $this->digest = $digest;
+        $this->isPredigested = $isPredigested;
+    }
+
+    public function setMimeType(?string $mimeType): void
+    {
+        if (isset($this->mimeType)) {
+            throw new \LogicException('Cannot set mime type: it was already set on the asset.');
+        }
+
+        $this->mimeType = $mimeType;
+    }
+
+    public function setContent(string $content): void
+    {
+        if (isset($this->content)) {
+            throw new \LogicException('Cannot set content: it was already set on the asset.');
+        }
+
+        $this->content = $content;
+    }
+
+    public function addDependency(MappedAsset $asset, bool $isLazy = false): void
+    {
+        $this->dependencies[] = new AssetDependency($asset, $isLazy);
+    }
+
+    public function getPublicPathWithoutDigest(): string
+    {
+        if ($this->isPredigested) {
+            return $this->publicPath;
+        }
+
+        // remove last part of publicPath and replace with last part of logicalPath
+        $publicPathParts = explode('/', $this->publicPath);
+        $logicalPathParts = explode('/', $this->logicalPath);
+        array_pop($publicPathParts);
+        $publicPathParts[] = array_pop($logicalPathParts);
+
+        return implode('/', $publicPathParts);
+    }
+
+    public static function extractExtension(string $logicalPath): string
```

Comment:
can't we drop this method and use the native call instead? less maintenance for us